Calendar/Get-CalendarDiagnosticObjectsSummary.ps1

```diff
@@ -0,0 +1,922 @@
+                    }
+                    default {
+                        [array] $Output = "[$($CalLog.ResponsibleUser)] $($CalLog.TriggerAction) the Meeting with $($CalLog.Client).";
+                        [bool] $MeetingSummaryNeeded = $False;
+                    }
+                }
+            }
+            Canceled {
+                [array] $Output = "$($CalLog.ResponsibleUser) Created a new Cancellation message for the $($CalendarItemTypes.($CalLog.ItemClass)) with $($CalLog.Client)";
+            }
+            default {
+                if ($CalLog.TriggerAction -eq "Create") {
+                    $Action = "New";
+                } else {
+                    $Action = "$($CalLog.TriggerAction)";
+                }
+                [array] $Output = "$($Action) was performed on the $($CalLog.ItemClass) by $($CalLog.ResponsibleUser) with $($CalLog.Client)";
+            }
+        }
+
+        $Time = "$($CalLog.LogRow) -- $($CalLog.LastModifiedTime)"
+
+        if ($Output) {
+            if ($MeetingSummaryNeeded) {
+                MeetingSummary -Time $Time -MeetingChanges $Output;
+                MeetingSummary -Time " " -ShortVersion -Entry $CalLog;
+            } else {
+                MeetingSummary -Time $Time -MeetingChanges $Output;
+                if ($AddChangedProperties) {
+                    ChangedProperties;
+                }
+            }
+        }
+
+        if ($CalendarItemTypes.($CalLog.ItemClass) -eq "IpmAppointment" -or $CalendarItemTypes.($CalLog.ItemClass) -eq "ExceptionMsgClass") {
+            $PreviousCalLog = $CalLog;
+        }
+    }
+
+    $Results = @();
+}
+
+$GlobalObjectId = @();
+
+foreach ($ObjectId in $GetCDO.CleanGlobalObjectId) {
+    if (![string]::IsNullOrEmpty($ObjectId) -and $ObjectId -ne "NotFound" -and $ObjectId -ne "InvalidSchemaPropertyName" -and $ObjectId.length -ge 90) {
+        $GlobalObjectId += $ObjectId;
+    }
+}
+
+$UniqueMeetingID = $GlobalObjectId | Select-Object -Unique;
+
+if ($UniqueMeetingID.count -gt 1) {
+    $UniqueMeetingID | ForEach-Object {
+        $MeetingID = $_;
+        $script:GCDO = Get-CalendarDiagnosticObjects -Identity $Identity -MeetingID $MeetingID -CustomPropertyNames AppointmentCounterProposal, AppointmentRecurring, CalendarItemType, CalendarProcessed, ClientIntent, DisplayAttendeesCc, DisplayAttendeesTo, EventEmailReminderTimer, ExternalSharingMasterId, FreeBusyStatus, From, HasAttachment, IsAllDayEvent, IsCancelled, IsMeeting, MapiEndTime, MapiStartTime, OnlineMeetingConfLink, OnlineMeetingExternalLink, OnlineMeetingInternalLink, SentRepresentingDisplayName, SentRepresentingEmailAddress -WarningAction Ignore -MaxResults 2000;
+        BuildCSV;
+        BuildTimeline;
+    }
+} elseif ($UniqueMeetingID.count -eq 1) {
+    $GCDO = $GetCDO;
+    $GetCDO = @();
+    BuildCSV;
+    BuildTimeline;
+} else {
+    Write-Warning "A valid meeting ID was not found, manually confirm the meetingID";
+}
+$TimeSum = @();
```

Comment:
This needs to be defined prior to being used in `Convert-Data` which is called by `BuildTimeline->MeetingSummary->Convert-Data`. So this should be declared prior to `BuildTimeline` is called.

Reply:
Removed, was not needed